src/transaction/locator_sr.c

```diff
@@ -13826,9 +13829,23 @@ locator_multi_insert_force (THREAD_ENTRY * thread_p, HFID * hfid, OID * class_oi
   if (error_code != NO_ERROR)
     {
       ASSERT_ERROR ();
-      return error_code;
+      goto cleanup;
     }
 
-  return NO_ERROR;
+cleanup:
+  // Check if we have errors to abort the sysop.
+  if (error_code != NO_ERROR)
+    {
+      // Safeguard
+      ASSERT_ERROR ();
+      log_sysop_abort (thread_p);
+    }
+  else
+    {
+      // Commit the sysop
+      log_sysop_commit (thread_p);
```

Comment:
Actually sysop commit is not right here. Records should be removable in case of rollback/undo.

Reply:
it should be `log_sysop_attach_to_outer`.

Reply:
it is actually more complicated than that... insert records should be undone, but undoing heap pages allocations & linking in heap is not so straight-forward.

what happens if a scan is in progress?

Reply:
hmm...  We are in a trouble for the case. 

Then, no other option but the current?

Reply:
None of the simple solutions - commit or attach to outer - is satisfying. Unfortunately, I can only think of non-trivial solutions. For instance, ending with a logical undo, that on abort deletes all inserted records.

We may discuss the matter with the whole team and maybe we identify a better solution.